packages/core/src/redux/slices/breakout.ts

```diff
@@ -0,0 +1,140 @@
+    ...createBreakout(),
+    groupId: null,
+};
+
+export const breakoutSlice = createSlice({
+    name: "breakout",
+    initialState,
+    reducers: {},
+    extraReducers: (builder) => {
+        builder.addCase(signalEvents.roomJoined, (state, action) => {
+            if (action.payload.breakout) {
+                return {
+                    ...state,
+                    ...createBreakout(action.payload.breakout),
+                };
+            }
+            return state;
+        });
+        builder.addCase(signalEvents.breakoutSessionUpdated, (state, action) => {
+            return {
+                ...state,
+                ...createBreakout(action.payload),
+            };
+        });
+        builder.addCase(signalEvents.breakoutGroupJoined, (state, action) => {
+            // @ts-ignore
+            if (action.meta?.localParticipantId !== action.payload.clientId) {
+                return state;
+            }
+            return {
+                ...state,
+                groupId: action.payload.group,
+            };
+        });
+    },
+});
+
+export const doBreakoutJoin = createAppThunk((payload: { group: string }) => (_, getState) => {
+    const state = getState();
+
+    const { socket } = selectSignalConnectionRaw(state);
+    socket?.emit("join_breakout_group", { group: payload.group });
+});
+
+/**
+ * Selectors
+ */
+export const selectBreakoutRaw = (state: RootState) => state.breakout;
+export const selectBreakoutInitiatedBy = (state: RootState) => state.breakout.initiatedBy;
+export const selectBreakoutActive = (state: RootState) => !!state.breakout.startedAt;
+export const selectBreakoutAssignments = (state: RootState) => state.breakout.assignments;
+export const selectBreakoutGroups = (state: RootState) => state.breakout.groups;
+
+export const selectBreakoutCurrentId = createSelector(
+    selectBreakoutRaw,
+    selectLocalParticipantBreakoutGroup,
+    (raw, localParticipantBreakoutGroup) => {
+        return raw.groupId || localParticipantBreakoutGroup || "";
+    },
+);
+
+export const selectBreakoutCurrentGroup = createSelector(
+    selectBreakoutRaw,
+    selectBreakoutCurrentId,
+    (raw, breakoutCurrentId) => {
+        const name = raw.groups?.[breakoutCurrentId];
+        if (!name) return null;
+        return { id: breakoutCurrentId, name };
+    },
+);
+
+startAppListening({
+    actionCreator: signalEvents.breakoutMoveToGroup,
+    effect: (_, { dispatch, getState }) => {
+        const state = getState();
+        const localParticipant = selectLocalParticipantRaw(state);
+        const breakoutGroupAssigned = localParticipant.breakoutGroupAssigned;
+
+        if (breakoutGroupAssigned) {
+            dispatch(doBreakoutJoin({ group: breakoutGroupAssigned }));
+        }
+    },
+});
+
+startAppListening({
+    actionCreator: signalEvents.breakoutMoveToMain,
+    effect: (_, { dispatch }) => {
+        dispatch(doBreakoutJoin({ group: "" }));
+    },
+});
```

Comment:
I found myself in a bit of a mixed up state when:

The SDK client is in a breakout group
The breakout session ends when autoMoveToMain is NOT enabled, so the client in the breakout group won't get sent back to the main room automatically The SDK client ends up stuck in that sort of in between state of being in a breakout group even after the sesson has ended.
Maybe we could add something to check for the breakout_updated event with active === false, and then send back to the room if that automatic setting isn't enabled?

```suggestion
startAppListening({
    actionCreator: signalEvents.breakoutSessionUpdated,
    effect: (action, { dispatch, getState }) => {
        const state = getState();

        if (action.payload?.initiatedBy?.active === false) {
            if (state.breakout.autoMoveToMain === false) {
                dispatch(doBreakoutJoin({ group: "" }));
            }
        }
    },
});

});
```

Reply:
Ah, yes, good catch. This is actually how it behaves in the pwa - we don't automatically push people back to the main room after the breakout session ends. But there we have some UI hints. Probably the best option is to auto-push the SDK clients as you say. Will add this 👍

Reply:
Pushed this now. Also added a listener in `localMedia` slice that will mute the microphone when the breakout session ends